Modify integration configuration to explicitly use intel MPI and intel compiler

cmcantalupo opened the pull request.

bgeltz commented on integration/config/mcfly_env.sh:
```diff
@@ -0,0 +1,52 @@
+export MPIFC=mpiifort
+export MPIF77=mpiifort
+export MPIF90=mpiifort
+export FI_PROVIDER=psm2
```
```suggestion
```

Why is this necessary?

bgeltz commented on integration/config/mcfly_env.sh:
```diff
@@ -0,0 +1,52 @@
+# This script is intended to be sourced within an existing script or shell ONLY.
+# It is NOT intended to be ./ executed.
+
+module swap mvapich2 impi >& /dev/null || true
```
```suggestion
```

I'm going to change the cluster wide defaults if this is necessary.

cmcantalupo replied:
Not the best solution as written anyway (no check for impi being loaded in the end)

bgeltz reviewed (changes requested):
I thought these changes were going to go into build_env.sh.

When we wrote build_env.sh originally, it was written for the default McFly environment.

cmcantalupo committed:
Switch default build environment for integration to Intel

- The latest version of the intel compiler wrappers use the
  gcc compiler when mpicc is specified rather than icc
- This effects the C++ and fortran wrappers similarly.
- The default config now explicitly requests the intel
  mpi compiler wrappers wrapping the intel compiler.

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>
